Release v2 of Effection 🥳🥳🥳

jnicklas opened the pull request:
## Motivation

We want to actually ship something

## Approach

![ship it squirrel](https://user-images.githubusercontent.com/134/136370648-76f22dcb-c05a-44f2-a215-260f67d86615.png)

### Alternate Designs

Don't ship it

### Possible Drawbacks or Risks

We have to live with it as is for a while

### TODOs and Open Questions
- [ ] Should we ship it?

jnicklas committed:
Remove prerelease changes

jnicklas committed:
Add 2.0 changeset

jnicklas committed:
Release 2.0.0

cowboyd commented on packages/atom/package.json:
```diff
@@ -41,9 +41,9 @@
-    "@effection/stream": "2.0.0-beta.6",
+    "@effection/channel": "^2.0.0",
+    "@effection/core": "^2.0.0",
+    "@effection/stream": "^2.0.0",
```
This might be a good time to change the "contrib" packages to use "effection" directly.

jnicklas replied:
Yes, we might want to do that. I'm still not sure which option is better 🤷‍♂️

cowboyd reviewed (commented):
![Jeremiah Johnson](https://media.giphy.com/media/qWsU3ILSNlPZ6/giphy.gif)

cowboyd committed:
update dependencies for core / non-core packages

1. "core" packages pin each other
2. "contrib" packages ^ depend on "effection" and "effection" package only
2. "inspect" packages pin each other and ^depend on "effection" and "contribs"

cowboyd committed:
Don't build mocha until after "effection" package

cowboyd committed:
remove process reference from main tsconfig

`@effection/main` depends on process at test-time, but not at
build-time.
